File: src/StickedWords.Domain/Models/UserInfo.cs
namespace StickedWords.Domain.Models;

public record UserInfo
{
    public required string Login { get; init; }

    public required string? Surname { get; init; }

    public required string? GivenName { get; init; }

    public required string? Email { get; init; }

    public required string ExternalId { get; init; }

    public required string AuthProvider { get; init; }
}

File: src/StickedWords.UI.React/src/models/UserInfo.ts
export class UserInfo {
  login!: string;
  surname?: string;
  givenName?: string;
  email?: string;
  externalId!: string;
  authProvider!: string;

  static fromJson(json: any): UserInfo {
    const result = new UserInfo();

    result.login = json.login ?? "";
    result.surname = json.surname ?? null;
    result.givenName = json.givenName ?? null;
    result.email = json.email ?? null;
    result.externalId = json.externalId ?? "";
    result.authProvider = json.authProvider ?? "";

    return result;
  }
}